src/components/AppProvider/AppProvider.tsx

```diff
@@ -35,82 +39,54 @@ export interface AppProviderProps extends AppBridgeOptions {
   linkComponent?: LinkLikeComponent;
   /** Custom logos and colors provided to select components */
   theme?: Theme;
+  /** Inner content of the application */
+  children: OnlyChild;
 }
 
-export class AppProvider extends React.Component<AppProviderProps, State> {
-  private stickyManager: StickyManager;
-  private scrollLockManager: ScrollLockManager;
-  private uniqueIdFactory: UniqueIdFactory;
-
-  constructor(props: AppProviderProps) {
-    super(props);
-    this.stickyManager = new StickyManager();
-    this.scrollLockManager = new ScrollLockManager();
-    this.uniqueIdFactory = new UniqueIdFactory(globalIdGeneratorFactory);
+export function AppProvider({
+  apiKey,
+  shopOrigin,
+  forceRedirect,
+  i18n,
+  linkComponent,
+  theme = {logo: null},
+  children,
+}: AppProviderProps) {
+  const stickyManager = useRef(new StickyManager());
+  const scrollLockManager = useRef(new ScrollLockManager());
+  const uniqueIdFactory = useRef(new UniqueIdFactory(globalIdGeneratorFactory));
 
-    const {i18n, apiKey, shopOrigin, forceRedirect, linkComponent} = this.props;
+  const [appBridge, setAppBridge] = useState(
+    createAppBridge({shopOrigin, apiKey, forceRedirect}),
+  );
 
-    // eslint-disable-next-line react/state-in-constructor
-    this.state = {
-      link: linkComponent,
-      intl: new I18n(i18n),
-      appBridge: createAppBridge({shopOrigin, apiKey, forceRedirect}),
-    };
-  }
-
-  componentDidMount() {
+  useEffect(() => {
+    console.log('useEffect 1 MOUNT');
     if (document != null) {
-      this.stickyManager.setContainer(document);
+      stickyManager.current.setContainer(document);
     }
-  }
-
-  componentDidUpdate({
-    i18n: prevI18n,
-    linkComponent: prevLinkComponent,
-    apiKey: prevApiKey,
-    shopOrigin: prevShopOrigin,
-    forceRedirect: prevForceRedirect,
-  }: AppProviderProps) {
-    const {i18n, linkComponent, apiKey, shopOrigin, forceRedirect} = this.props;
-
-    if (
-      i18n === prevI18n &&
-      linkComponent === prevLinkComponent &&
-      apiKey === prevApiKey &&
-      shopOrigin === prevShopOrigin &&
-      forceRedirect === prevForceRedirect
-    ) {
-      return;
-    }
-
-    // eslint-disable-next-line react/no-did-update-set-state
-    this.setState({
-      link: linkComponent,
-      intl: new I18n(i18n),
-      appBridge: createAppBridge({shopOrigin, apiKey, forceRedirect}),
-    });
-  }
+  }, []);
 
-  render() {
-    const {theme = {logo: null}, children} = this.props;
-    const {intl, appBridge, link} = this.state;
+  useEffect(() => {
+    console.log('useEffect 2 MOUNT');
```

Comment:
Remove these console logs... maybe look into preventing this effect on first mount!